eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,57 @@
+function Get-Namespaces-From-DLL($dllPath) {
+    $file = [System.IO.File]::OpenRead($dllPath)
+    try {
+        # Use to parse the namespaces out from the dll file.
+        $pe = [System.Reflection.PortableExecutable.PEReader]::new($file)
+        try {
+            $meta = [System.Reflection.Metadata.PEReaderExtensions]::GetMetadataReader($pe)
+            foreach ($typeHandle in $meta.TypeDefinitions) {
+                $type = $meta.GetTypeDefinition($typeHandle)
+                $attr = $type.Attributes
+                if ($attr -band 'Public' -and !$type.IsNested) {
+                    [pscustomobject]@{
+                        Name = $meta.GetString($type.Name)
+                        Namespace = $meta.GetString($type.Namespace)
+                    }
+                }
+            }
+        } finally {
+            $pe.Dispose()
+        }
+    } finally {
+        $file.Dispose()
+    }
+}
+
+function Fetch-Namespaces-From-Dotnet-Nupkg ($nupkgFilePath, $destination) {
+    $tempLocation = (Join-Path ([System.IO.Path]::GetTempPath()) "dotnetNupkg")
+    if (Test-Path $tempLocation) {
+        Remove-Item $tempLocation/* -Recurse -Force 
+    }
+    else {
+        New-Item -ItemType Directory -Path $tempLocation -Force | Out-Null
+    }
+
+    Write-Host "Unzipping nupkg..."
+    Write-Host $nupkgFilePath
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($nupkgFilePath, $tempLocation)
+    # .NET core includes multiple target framework. We currently have the same namespaces for different framework. 
+    # Will use whatever the first dll file.
+    Write-Host "Searching ddl file..."
+    $firstDllFiles = Get-ChildItem "$tempLocation/lib" -Filter '*.dll' -Recurse
+    if (!$firstDllFiles) {
+        Write-Error "Can't find any dll file from dotnet $nupkgFilePath."
+        return
+    }
+    Write-Host "Dll file found: $($firstDllFiles[0].FullName)"
+    $namespaces = Get-Namespaces-From-DLL $firstDllFiles[0].FullName
```

Comment:
Nit: PowerShell functions/scripts are intended to follow a "{Verb}-{Noun}" format with only a single dash. This is why coloring is off here. This is enforced in some places but not in scripts like these. Still, for maximum portability I recommend following the intended pattern with script file names and function/cmdlet names e.g. `Get-NamepspacesFromDll`.